posthog/api/signup.py

```diff
@@ -30,16 +30,27 @@
 
 
 def verify_email_or_login(request: HttpRequest, user: User) -> None:
-    require_verification_feature = posthoganalytics.feature_enabled("require-email-verification", str(user.distinct_id))
-    if is_cloud() and require_verification_feature:
+    require_verification_feature = (
+        posthoganalytics.get_feature_flag(
+            "require-email-verification", str(user.distinct_id), person_properties={"email": user.email}
```

Comment:
I would say it's expected as you need to match against the set of filters you set on the FF?

Reply:
It should always evaluate correctly!

The problem with not setting the email here might be that for the given user's distinct ID, we may not yet have ingested events, so the database doesn't have their email ID, in which case we evaluate to false.

Having an email ID here ensures we use this property itself, vs. relying on what's in the DB.

Most likely this is what's happening, since this is happening too close to signup, and ingestion delay can be upto 5 minutes in the usual case.